README.rst

```diff
@@ -112,6 +112,42 @@ If you want to make an RPC and continue an existing trace, you can inject the cu
             request.add_header(k,v)
         ... # make request
 
+Using tracing hooks
+===================
+
+You can specify start/finish hooks to be called whenever the middleware
+creates/finishes a new Span.  They can act as an integration point to interact
+with the spans handled by this Django middleware.  They are python callables
+and can be provided via the django settings using the `OPENTRACING_HOOKS`
+options. For example:
+
+.. code-block:: python
+
+    OPENTRACING_HOOKS = {'start': start_hook, 'finish': finish_hook}
+
+
+- The start hook expects the newly created span as argument.
+- The finish_hook expects the span that is ready to be finished.
+
+Any of them can be omitted.
+
+A hook can return a 'True' value and let the middleware know that it should not
+handle the Span.
+
+- If the 'start_hook' returns true, then the middleware does not keep
+  any reference to the Span.
+
+- If the 'finish_hook' returns true, then the middleware does not try
+  to finish the span in hands.
+  Note that based on the return value of the start hook, a reference to
+  a Span for the given request may not even exist, but the finish hook
+  will be called with a `None` argument.
+
```

Comment:
Personally I think returning `False` for both hooks is the one that should be causing the `Span` to not be handled. Also, I honestly think that, if we don't handle the `Span` for `start_hook`, we should do some extra effort and *not* invoke the `finish_hook` at all.

That being said: I remember more than a few people asking for these kind of hooks, so this is definitely something nice to have - however, I don't remember anybody wanting to *discard* `Span` instances from these.

Reply:
If I remember correctly, I believe that I had allowed `finish_hook` to be called with `None`, to notify the external Span manager that the span of the current request/current context has finished, even though `python-django` does not know anything about how the external entity handles/tracks the Span. This allows to completely externalize the span management, but still provide notifications about the start and finish events of the current request's span.

Reply:
Hey. To me it sounds you want to do the active `Span` management yourself, outside the middleware, which sounds... complicated. Observe we will be issuing soon the [OT 2.0 API ](https://github.com/opentracing/opentracing-python/issues/65)(which includes active `Span` management), which may even remove the need for `_current_span`.

Would that solve your problem? (Observe I'd like to keep the hooks, but without the extra `Span` handling/not handling, if possible).

Reply:
@carlosalberto what do you think?

I would argue pretty strongly against doing active Span management in this PR – that is what the 2.0 API is _for_, and it will lead to extra code, bugs, or both. :-/

Reply:
@bhs Oh, definitely, lets not mix them.

@philipgian Sorry for the late answer - so, yes, I think we could update this, removing the `Span` handling part, but keeping the (useful) hooks - POST OT 2.0, me/we can review it in case you still need any of the more advanced scenario, and put it back if still needed. Think that would work?

Let me know, thanks ;)

Reply:
@carlosalberto Sounds good to me :)